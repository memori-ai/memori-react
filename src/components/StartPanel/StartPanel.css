.memori--gamification-badge {
    position: absolute;
    right: 0.25rem;
    bottom: -1.25rem;
    display: flex;
    width: 2rem;
    height: 2rem;
    align-items: center;
    justify-content: center;
    border: 1.5px solid var(--memori-primary);
    border-radius: 50%;
    background: #fff;
    box-shadow: 0 0 5px 6px rgba(255, 255, 255, 0.5);
}

.memori--gamification-badge span {
    font-size: 1.2rem;
}

.memori--start-panel {
    width: 100%;
    align-self: flex-start;
    padding: var(--memori-inner-content-pad, 0);
    border-radius: 10px;
    margin-top: 2rem;
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: var(--memori-inner-bg, transparent);
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
}

.memori--start-panel .memori--start-panel-title {
    color: var(--memori-text-color, var(--memori-primary)) !important;
}

.memori--start-panel .memori--start-panel-description {
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85)) !important;
}

@media (max-width: 870px) {
    .memori--start-panel {
        margin-top: 1rem;
    }
}

.memori--cover {
    position: relative;
    width: 100%;
    padding-top: 31.25%;
    border-radius: 3px;
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
}

.memori--avatar {
    position: relative;
    z-index: 0;
    top: -20px;
    display: inline-flex;
    overflow: hidden;
    width: 60px;
    height: 60px;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    margin-right: 10px;
    background: #fff;
    box-shadow: 0 0 5px rgba(50, 50, 50, 0.3);
}

.memori--avatar img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

.memori--title {
    display: inline-block;
    margin: 0;
    color: var(--memori-primary);
    font-size: 24px;
    font-weight: 600;
    line-height: 1.8;
}

.memori--description,
.memori--needsPosition {
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85));
    font-size: 14px;
    line-height: 1.6;
}

.memori--description-text {
    display: block;
}

.memori--translation-toggle {
    height: auto;
    padding-right: 0;
    padding-left: 0;
    color: var(--memori-primary);
    font-size: 0.85em;
}

.memori--start-description {
    margin-top: 0.5rem;
    font-size: 0.85em;
    opacity: 0.85;
}

.memori--start-button {
    margin-right: 1em;
}

.memori--language-chooser {
    margin-bottom: 1rem;
}

.memori--language-chooser svg {
    margin-right: 0.5rem;
}

.memori--language-chooser select,
.memori--language-chooser .memori--select {
    min-width: 7rem;
    margin-left: 0.5rem;
}

.memori--language-chooser select option {
    background: #fff;
    color: #000;
}

.memori--completions-enabled,
.memori--board-of-experts,
.memori--nsfw {
    position: absolute;
    right: 0.25rem;
    bottom: -1.25rem;
    display: flex;
    width: 2rem;
    height: 2rem;
    align-items: center;
    justify-content: center;
    border: 1.5px solid var(--memori-primary);
    border-radius: 50%;
    background: #fff;
    box-shadow: 0 0 5px 6px rgba(255, 255, 255, 0.5);
    color: #000;
}

.memori--completions-enabled span {
    font-size: 1.2rem;
}

.memori--completions-enabled svg {
    overflow: visible;
    width: 1rem;
    height: 1rem;
}

.memori--board-of-experts + .memori--nsfw,
.memori--completions-enabled + .memori--nsfw {
    right: 2.5rem;
}

.memori--board-of-experts svg {
    overflow: visible;
    width: 1rem;
    height: 1rem;
    fill: #000;
}

.memori--deep-thought-disclaimer {
    position: relative;
    padding: 1em 1.5em;
    border: 2px solid var(--memori-warning-color, #faad14);
    border-radius: var(--memori-border-radius, 10px);
    margin-top: 1rem;
    font-size: 0.8em;
}

.memori--deep-thought-disclaimer::before {
    position: absolute;
    z-index: -1;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: var(--memori-warning-color, #faad14);
    content: '';
    opacity: 0.2;
}

.memori--deep-thought-disclaimer h2 {
    margin-top: 0;
    margin-bottom: 0.5em;
    color: var(--memori-text-color, rgba(0, 0, 0, 0.85));
}

.memori--deep-thought-disclaimer .memori-tooltip {
    position: relative;
    margin-bottom: 1rem;
    margin-left: 1rem;
    float: right;
}

.memori--deep-thought-disclaimer .memori-tooltip .memori-tooltip--trigger svg {
    width: 1.5rem;
    height: 1.5rem;
    padding: 5px;
    border: 1px solid #666;
    border-radius: 50%;
}

.memori--deep-thought-disclaimer a {
    color: inherit;
    text-decoration: underline;
}
