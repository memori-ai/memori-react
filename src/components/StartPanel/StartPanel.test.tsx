import React from 'react';
import { render } from '@testing-library/react';
import { memori, tenant, sessionID, integration, user } from '../../mocks/data';
import StartPanel from './StartPanel';

it('renders StartPanel unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={memori}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with completions enabled unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        enableCompletions: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel for board of experts unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        enableBoardOfExperts: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with deep thought enabled unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        enableDeepThought: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      isUserLoggedIn
      user={{
        ...user,
        pAndCUAccepted: true,
      }}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with deep thought but unlogged unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        enableDeepThought: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      showLogin
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with deep thought enabled but without permission flag unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        enableDeepThought: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      isUserLoggedIn
      user={{
        ...user,
        pAndCUAccepted: false,
      }}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel on instruct unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={memori}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={true}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with position required unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        needsPosition: true,
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={true}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with integrationConfig unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={memori}
      tenant={tenant}
      integrationConfig={JSON.parse(integration.customData ?? '{}')}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with multilangual unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={memori}
      tenant={tenant}
      isMultilanguageEnabled
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders StartPanel with completion provider down unchanged', () => {
  const { container } = render(
    <StartPanel
      memori={{
        ...memori,
        completionConfigForQuestionAnswering: {
          completionConfigID: '1',
          configName: 'openai-gpt-4',
          provider: 'OpenAI',
          model: 'gpt-4',
        },
      }}
      tenant={tenant}
      language="it"
      userLang="en"
      setUserLang={() => {}}
      openPositionDrawer={() => {}}
      instruct={false}
      sessionId={sessionID}
      clickedStart={false}
      onClickStart={() => {}}
      setShowLoginDrawer={jest.fn()}
      _TEST_forceProviderStatus="major_outage"
    />
  );
  expect(container).toMatchSnapshot();
});
