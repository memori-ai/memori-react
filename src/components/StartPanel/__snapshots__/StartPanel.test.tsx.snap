// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StartPanel on instruct unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--gamification-badge"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            gamification.level üåç, 61 gamification.points
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="gamification.level üåç, 61 gamification.points"
            >
              üåç
            </span>
          </div>
        </div>
      </div>
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--description"
    >
      <p>
        <span
          class="memori--description-text"
        >
          Lorem ipsum.
        </span>
      </p>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        write_and_speak.instructButton
      </button>
      <p
        class="memori--start-description"
      >
        write_and_speak.pageInstructExplanation
      </p>
    </div>
  </div>
</div>
`;

exports[`renders StartPanel unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--description"
    >
      <p>
        <span
          class="memori--description-text"
        >
          Lorem ipsum.
        </span>
      </p>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        write_and_speak.tryMeButton
      </button>
      <p
        class="memori--start-description"
      >
        write_and_speak.pageTryMeExplanation
      </p>
    </div>
  </div>
</div>
`;

exports[`renders StartPanel with completions enabled unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--description"
    >
      <p>
        <span
          class="memori--description-text"
        >
          Lorem ipsum.
        </span>
      </p>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        write_and_speak.tryMeButton
      </button>
      <p
        class="memori--start-description"
      >
        write_and_speak.pageTryMeExplanation
      </p>
    </div>
  </div>
</div>
`;

exports[`renders StartPanel with gamification level unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--gamification-badge"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            gamification.level üåç, 61 gamification.points
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="gamification.level üåç, 61 gamification.points"
            >
              üåç
            </span>
          </div>
        </div>
      </div>
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--description"
    >
      <p>
        <span
          class="memori--description-text"
        >
          Lorem ipsum.
        </span>
      </p>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        write_and_speak.tryMeButton
      </button>
      <p
        class="memori--start-description"
      >
        write_and_speak.pageTryMeExplanation
      </p>
    </div>
  </div>
</div>
`;

exports[`renders StartPanel with integrationConfig unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--description"
    >
      <p>
        <span
          class="memori--description-text"
        >
          Lorem ipsum.
        </span>
      </p>
      <div
        class="memori--language-chooser"
      >
        <label
          for="user-lang-pref"
          id="user-lang-pref-label"
        >
          write_and_speak.iWantToTalkToIn
        </label>
        <select
          aria-labelledby="user-lang-pref-label"
          class="memori-select--button"
          id="user-lang-pref"
        >
          <option
            aria-label="Italiano"
            value="IT"
          >
            Italiano
          </option>
          <option
            aria-label="English"
            value="EN"
          >
            English
          </option>
          <option
            aria-label="Deutsch"
            value="DE"
          >
            Deutsch
          </option>
          <option
            aria-label="Espa√±ol"
            value="ES"
          >
            Espa√±ol
          </option>
          <option
            aria-label="Fran√ßais"
            value="FR"
          >
            Fran√ßais
          </option>
          <option
            aria-label="Portugu√™s"
            value="PT"
          >
            Portugu√™s
          </option>
          <option
            aria-label="–£–∫—Ä–∞—ó–Ω—Å—å–∫–∞"
            value="UK"
          >
            –£–∫—Ä–∞—ó–Ω—Å—å–∫–∞
          </option>
          <option
            aria-label="–†—É—Å—Å–∫–∏–π"
            value="RU"
          >
            –†—É—Å—Å–∫–∏–π
          </option>
          <option
            aria-label="Polski"
            value="PL"
          >
            Polski
          </option>
        </select>
      </div>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        write_and_speak.tryMeButton
      </button>
      <p
        class="memori--start-description"
      >
        write_and_speak.pageTryMeExplanation
      </p>
    </div>
  </div>
</div>
`;

exports[`renders StartPanel with position required unchanged 1`] = `
<div>
  <div
    class="memori--start-panel"
  >
    <div
      class="memori--cover"
    >
      <div
        class="memori--gamification-badge"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            gamification.level üåç, 61 gamification.points
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="gamification.level üåç, 61 gamification.points"
            >
              üåç
            </span>
          </div>
        </div>
      </div>
      <div
        class="memori--completions-enabled"
      >
        <div
          class="memori-tooltip memori-tooltip--align-left"
        >
          <div
            class="memori-tooltip--content"
          >
            completionsEnabled
          </div>
          <div
            class="memori-tooltip--trigger"
          >
            <span
              aria-label="completionsEnabled"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-rule="evenodd"
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                    fill-rule="evenodd"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <picture
      class="memori--avatar"
    >
      <source
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
      <img
        alt="Memori"
        src="https://app.twincreator.com/images/twincreator/square_logo.png"
      />
    </picture>
    <h2
      class="memori--title"
    >
      Memori
    </h2>
    <div
      class="memori--needsPosition"
    >
      <p>
        write_and_speak.requirePosition
      </p>
      <button
        class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
      >
        widget.position
      </button>
    </div>
  </div>
</div>
`;
